homeconnect/api.py

```diff
@@ -414,14 +430,30 @@ def get_settings(self):
         self.status.update(self.json2dict(settings["settings"]))
         return self.status
 
+    def get_setting(self, settingkey):
+        """Get full information about a setting."""
+        setting = self.get(f"/settings/{settingkey}")
+        if not setting:
+            return {}
+        setting_dict = self.json2dict(setting)
+        self.status[settingkey] = setting_dict
+        return setting_dict
+
     def set_setting(self, settingkey, value):
         """Change the current setting of `settingkey`."""
         return self.put(
             "/settings/{}".format(settingkey),
             {"data": {"key": settingkey, "value": value}},
         )
 
-    def set_options_active_program(self, option_key, value, unit=None):
+    def set_options_active_program(self, options):
```

Comment:
This looks like a breaking change, can you please elaborate?

Reply:
I have designated the function name `set_options_X_program` for the function that sets multiple options simultaneously, as it contains the plural form of 'options'. 
Also for the function that will set a single option I have designated `set_option_X_program` as the function name.

I know that it could be a breaking change, but it would be consistent.

Perhaps an alternative approach for supporting both functionalities could that in the `set_option_X_program` function use the endpoint that sends multiple options simultaneously if the `option_key` variable is an instance of `list[dict[str, Any]]`:
```python
    def set_options_active_program(self, option_key, value, unit=None):
        """Change the options of the currently active program."""
        if isinstance(option_key, list[dict[str, Any]]):
            return self.put(
                "/programs/active/options",
                {"data": {"options": options}},
            )
        if unit is None:
            _unit = {}
        else:
            _unit = {"unit": unit}
        return self.put(
            f"/programs/active/options/{option_key}",
            {"data": {"key": option_key, "value": value, **_unit}},
        )
```